src/java/org/apache/cassandra/journal/Flusher.java

```diff
@@ -259,7 +259,7 @@ private void asyncFlushBatch(ActiveSegment<K>.Allocation alloc, Executor executo
             alloc.awaitFlush(journal.metrics.waitingOnFlush);
             pending.decrementAndGet();
             written.incrementAndGet();
-            executor.execute(onDurable);
+            executor.execute(callback);
```

Comment:
rather than `AsyncWriteCallback extends Runnable` it might be better to do `callback::onSuccess`

Reply:
the issue I face is that we then have 3 methods, if the user defines `run` then `onSuccess` will never trigger; which is confusing.

A compromise is something like the following

```
interface Callback extends Runnable {
  void onFailure(Throwable t);
}
```

This makes it so that there are only 2 methods: success and failure

Reply:
Good point, agreed (and implemented). Cheers.